common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapConfigStoreTaskProvider.java

```diff
@@ -164,9 +166,20 @@ protected void findSecurityInstanceId(Link[] objectLinks) {
 
     protected void findServerInstanceId(BootstrapReadResponse readResponse) {
         this.serverInstances = new HashMap<>();
-        ((LwM2mObject) readResponse.getContent()).getInstances().values().forEach(instance -> {
-            serverInstances.put(((Long) instance.getResource(0).getValue()).intValue(), instance.getId());
-        });
+        try {
+            ((LwM2mObject) readResponse.getContent()).getInstances().values().forEach(instance -> {
+                var shId = OPAQUE.equals(instance.getResource(0).getType()) ? new BigInteger((byte[]) instance.getResource(0).getValue()).intValue() : instance.getResource(0).getValue();
+                int shortId;
+                if (shId instanceof Long) {
+                    shortId = ((Long) shId).intValue();
+                } else {
+                    shortId = ((Integer) shId).intValue();
```

Comment:
If the value is not Integer (Boolean or other types)?

Reply:
Only Long, Integer, int